### PR TITLE
`ci-operator-configresolver`: guarantee unique metadata when merging configs from multiple sources

### PR DESCRIPTION
**Pull request by smg247**
Now that we allow PRs from the same repo to be tested together, we have situations where we are merging multiple of the same configs together. In practice, this appears to be okay with `ci-operator`, but it is a waste of time and resources. Instead, we should not merge duplicated metadata.

For: https://issues.redhat.com/browse/DPTP-3691

### CODE REVIEW

File: pkg/registry/server/server.go [hunk 1]
```diff
@@ -461,8 +462,8 @@ func MetadataEntriesFromQuery(w http.ResponseWriter, r *http.Request) ([]api.Met
 		if variantsExist {
 			element.Variant = variants[i]
 		}
-		metadata = append(metadata, element)
+		metadata.Insert(element)
 	}
 
-	return metadata, nil
+	return metadata.UnsortedList(), nil
```
Comment (danilo-gemoli) [inline pkg/registry/server/server.go:468]: Ok, it's not sorted anymore but it doesn't matter.

Comment (smg247) [reply pkg/registry/server/server.go:468]: yes, the order doesn't matter. We don't have any unit tests here anyways. If we did, we would probably need to guarantee the order.

**Comment by smg247** [comment]
@danilo-gemoli can you retag this?

**Pull request merged**